### DISCUSSION TITLE
Use external library inside "Run JavaScript" activity

### DISCUSSION THREAD

**Discussion by truthz03** [elsa-workflows/elsa-core#3544] [Q&A] [answered]
Hi,

is it possible to use external JavaScript libraries inside the "Run JavaScript" activity?
This doesn't work:
![image](https://user-images.githubusercontent.com/14071920/208098614-03a2a61d-86ce-4add-8984-6c023f19ad49.png)

Thanks
Thomas

**Comment by abhishekZom** [comment, marked as answer]
Don't use import statement with the url. Instead use npm install and them import.

Can confirm it works, since I have used it

**Comment by truthz03** [comment]
Thanks for the answer.
I also found an alternative.
It is also possible to load the js file with a "SendHttpRequest" activity or save its content inside a variable.

After that it is possible to load it with "eval". 
With this method it is not needed to install it before.
![image](https://user-images.githubusercontent.com/14071920/208373985-56609404-3476-4703-a6db-b855e7bc00d0.png)

![image](https://user-images.githubusercontent.com/14071920/208373476-2b2e0bb5-5818-48df-a444-bf2900b42ed1.png)

**Reply by abhishekZom** [reply]
I would suggest not using eval. This is because the url can return harmful code via different kinds of attacks. And you don't have control over how its executed and it will execute un-sanitized code. Better to install it.

Read this from official docs:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval
